src/ziggurat/init.clj

```diff
@@ -73,6 +136,13 @@
 (defn validate-stream-routes [stream-routes]
   (s/validate StreamRoute stream-routes))
 
+(defn validate-modes [modes]
+  (let [invalid-modes (filter #(not (contains? (set (keys valid-modes-fns)) %)) modes)
+        invalid-modes-count (count invalid-modes)]
+    (when (pos? invalid-modes-count)
+      (throw (IllegalArgumentException. ^String (reduce (fn [acc mode]
```

Comment:
How can we better write this?

Reply:
Maybe use https://clojuredocs.org/clojure.core/ex-info.  eg: `(ex-info "Invalid modes passed." {:modes invalid-modes})`

Reply:
Okay. sure.